src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs

```diff
@@ -39,17 +39,17 @@ public DecompressionHandler(DecompressionMethods decompressionMethods, HttpMessa
 
         internal override async ValueTask<HttpResponseMessage> SendAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
         {
-            if (GZipEnabled && !request.Headers.AcceptEncoding.Contains(s_gzipHeaderValue))
+            if (GZipEnabled && request.Headers.AcceptEncoding.FirstOrDefault(encoding => string.Compare(encoding.Value, s_gzipHeaderValue.Value, StringComparison.OrdinalIgnoreCase) == 0) == null)
```

Comment:
I think string.Equal is better than string,Compare, since only care about equality